src/lwc/geTemplateBuilderBatchHeader/geTemplateBuilderBatchHeader.js

```diff
@@ -1,177 +1,245 @@
-import { LightningElement, track, api } from 'lwc';
-import getBatchFields from '@salesforce/apex/GE_TemplateBuilderCtrl.getBatchFields';
-import { findIndexByProperty, shiftToIndex, mutable, sort } from 'c/utilTemplateBuilder';
+import { LightningElement, track, api, wire } from 'lwc';
+import { findIndexByProperty, mutable, sort, dispatch, handleError } from 'c/utilTemplateBuilder';
+import { getObjectInfo } from 'lightning/uiObjectInfoApi';
+import DATA_IMPORT_BATCH_OBJECT from '@salesforce/schema/DataImportBatch__c';
 
-const REQUIRED_FIELDS = [
+const ADDITIONAL_REQUIRED_FIELDS = [
     'Name'
 ];
-Object.freeze(REQUIRED_FIELDS);
+Object.freeze(ADDITIONAL_REQUIRED_FIELDS);
+
+const BLACKLISTED_FIELDS = [
+    'Batch_Process_Size__c',
+    'Run_Opportunity_Rollups_while_Processing__c',
+    'Donation_Matching_Behavior__c',
+    'Donation_Matching_Implementing_Class__c',
+    'Donation_Matching_Rule__c',
+    'Donation_Date_Range__c',
+    'Post_Process_Implementing_Class__c',
+    'OwnerId',
+];
+Object.freeze(BLACKLISTED_FIELDS);
+
+const SORTED_BY = 'required';
+const SORT_ORDER = 'desc';
+const PICKLIST = 'Picklist';
+const API_NAME = 'apiName';
+const NAMESPACE = 'npsp__';
 
 export default class geTemplateBuilderBatchHeader extends LightningElement {
     @track isLoading = true;
-    @track batchFields;
-    @track selectedBatchFields;
-    /* Public setter for the tracked property selectedBatchFields */
-    // TODO: Needs to be revisited, WIP tied to retrieving and rendering an existing template
-    @api
-    set selectedBatchFields(selectedBatchFields) {
-        this.selectedBatchFields = selectedBatchFields;
+    @track batchFields = [];
+    @api selectedBatchFields;
+    dataImportBatchInfo;
+    wireAdapterArg;
+
+    @wire(getObjectInfo, { objectApiName: DATA_IMPORT_BATCH_OBJECT })
+    wiredBatchDataImportObject({ error, data }) {
+        if (data) {
+            this.dataImportBatchInfo = data;
+            this.init();
+        } else if (error) {
+            console.log(error);
```

Comment:
Definitely! Leaving a todo comment for the handleError function because it needs a bit more work. Planning on improving it during the validations round of work items.